engine/src/main/java/io/camunda/zeebe/engine/state/instance/DbJobState.java

```diff
@@ -412,9 +412,7 @@ private void makeJobActivatable(final DirectBuffer type, final long key, final S
 
     jobTypeKey.wrapBuffer(type);
     jobKey.wrapLong(key);
-    if (!TenantOwned.DEFAULT_TENANT_IDENTIFIER.equals(tenantId)) {
-      tenantIdKey.wrapString(tenantId);
-    }
+    tenantIdKey.wrapString(TenantOwned.DEFAULT_TENANT_IDENTIFIER.equals(tenantId) ? "" : tenantId);
```

Comment:
❓ I don't understand why we need to use an empty string. Why don't we just set it to `<default>`. When activating a job we can check for old non-migrated ones if there is an empty string behind the key. For new ones with `<default>` we can just use that.

Reply:
The idea was to group all `<default>` tenant data under the empty string key. We can also do what you're suggesting, so instead of having a special case on all `<default>` data, we have a special case only on old data.

Do you have a strong preference on this @remcowesterhoud ?

Reply:
It's not very strong, but I'd say the more data is in the unified format the better 🤷

Reply:
I think I agree with @remcowesterhoud here. But to be honest I find it difficult to reason about.

Let's recap the idea and see if we can use `<default>` instead of `""`.

The `activatableColumnFamily` is structured as `[[type, key], tenant_id] => nil`. It will not be migrated by a migration task, so for old data we only have `[type, key] => nil`.

When reading this data (i.e. iterate), we do a whileEqualPrefix on the `[type]` part of the key only. The visitor then determines whether the entry has a key with tenant_id or not. If it doesn't, we assume the job belongs to the default tenant, otherwise we can use the specified tenant. The specified tenant might be `<default>`, that's not a problem.

When inserting this data, we insert it with a tenant id, either a custom one or the default one. However, previously inserted data may still be without tenantId. That does not matter to inserting. Note that this uses an upsert, but I don't think this matters perse.

When deleting this data, we delete it directly. This means we need to know whether or not this job exists in the old form or not. I'm not sure whether using `""` as the `tenantIdKey` would allow direct access to the old data. We should test this. 
- If `""` does allow direct access, then we should not suffix the default entries with the default tenant identifier
- If `""` does not allow direct access, then we may need to do an additional lookup with `exists()` or so

Reply:
We decided to use the `DbTenantAwareKey` class instead, and perform a migration on the `activatableColumnFamily`.